web/src/components/ResourceCard/ResourceCardActions.tsx

```diff
@@ -5,21 +5,37 @@ import * as S from './ResourceCard.styled';
 
 interface IProps {
   id: string;
+  canEdit: boolean;
   onDelete(): void;
+  onEdit(): void;
 }
 
-const ResourceCardActions = ({id, onDelete}: IProps) => {
-  const onClick = useCallback(
+const ResourceCardActions = ({id, canEdit = true, onDelete, onEdit}: IProps) => {
+  const onDeleteClick = useCallback(
     ({domEvent}) => {
       domEvent?.stopPropagation();
       onDelete();
     },
     [onDelete]
   );
 
+  const onEditClick = useCallback(
+    ({domEvent}) => {
+      domEvent?.stopPropagation();
+      onEdit();
+    },
+    [onEdit]
+  );
+
+  const menuItems = [];
```

Comment:
We try to follow an FP paradigm, which includes avoiding side effects like this.

You can achieve the same by doing:
```javascript
const menuItems = useMemo(() => {
   const defaultItems = [{key: 'delete', label: <span data-cy="test-card-delete">Delete</span>, onClick: onDeleteClick}];
   
   return shouldEdit ? [{key: 'edit', label: <span data-cy="test-card-edit">Edit</span>, onClick: onEditClick}, ...defaultItems] : defaultItems;
}, []);
```